### ISSUE TITLE
Lean Fork

### ISSUE THREAD

**Issue by jorahn** [jorahn/rookworld-rlvr#16]
@claude 
Create a new version of the key training code in a new folder in src/.
I suspect there is quite significant dead code or multiple implementations for the same functionality. That makes it hard to pin down issues like memory leakage during training.
Create a new implementation based on the absolutely necessary pieces of the existing code base. Observe
- the pretrained rookworld model (a small 124m gpt2 model) is post-trained with grpo on 24g GPUs. Don't use random parameter inits 
- the model knows to tasks, "P:" and "A:" prefixed. It needs ~144 tokens to generate valid answers, never reduce this too low, even on OOM. Make sure you only work with relevant prompts and completions (use the prepared rookworld dataset), not random prompts. Don't selfplay, only train on prompts from the dataset
- put the frozen model for generation on cuda:1, the model in training on cuda:0. ensure you detach tensors for computations on cpu. validate correctness for "P:" task from schema and using Stockfish to verify top5 best moves, centipawn value after these moves and the top1 best move - and Python-chess for correctness of the "A:" task. Use simple reward shaping rules
- keep the implementation minimal, no mixed precision, no huge config classes for every component, just lean an focused. The only exception is: I want extensive logging of data flowing through the training steps (prompts, completions, tensor shapes, memory usage, rewards computation, kl divergence, loss ...).
